packages/shared/sdk-client/src/LDClientImpl.events.test.ts

```diff
@@ -73,4 +73,86 @@ describe('sdk-client object', () => {
       }),
     );
   });
+
+  it('produces track events with data', async () => {
+    await ldc.identify(carContext);
+
+    ldc.track('the-event', { the: 'data' }, undefined);
+    expect(MockEventProcessor).toHaveBeenCalled();
+    expect(ldc.eventProcessor!.sendEvent).toHaveBeenNthCalledWith(
+      2,
+      expect.objectContaining<InputCustomEvent>({
+        kind: 'custom',
+        key: 'the-event',
+        context: expect.objectContaining({
+          contexts: expect.objectContaining({
+            car: { key: 'test-car' },
+          }),
+        }),
+        data: { the: 'data' },
+        samplingRatio: 1,
+        creationDate: expect.any(Number),
+      }),
+    );
+    expect(logger.warn).not.toHaveBeenCalled();
+  });
+
+  it('produces track events with a metric value', async () => {
+    await ldc.identify(carContext);
+
+    ldc.track('the-event', undefined, 12);
+    expect(MockEventProcessor).toHaveBeenCalled();
+    expect(ldc.eventProcessor!.sendEvent).toHaveBeenNthCalledWith(
+      2,
+      expect.objectContaining<InputCustomEvent>({
+        kind: 'custom',
+        key: 'the-event',
+        context: expect.objectContaining({
+          contexts: expect.objectContaining({
+            car: { key: 'test-car' },
+          }),
+        }),
+        metricValue: 12,
+        samplingRatio: 1,
+        creationDate: expect.any(Number),
+      }),
+    );
+    expect(logger.warn).not.toHaveBeenCalled();
+  });
+
+  it('produces track events with a metric value and data', async () => {
+    await ldc.identify(carContext);
+
+    ldc.track('the-event', { the: 'data' }, 12);
+    expect(MockEventProcessor).toHaveBeenCalled();
+    expect(ldc.eventProcessor!.sendEvent).toHaveBeenNthCalledWith(
+      2,
+      expect.objectContaining<InputCustomEvent>({
+        kind: 'custom',
+        key: 'the-event',
+        context: expect.objectContaining({
+          contexts: expect.objectContaining({
+            car: { key: 'test-car' },
+          }),
+        }),
+        metricValue: 12,
+        data: { the: 'data' },
+        samplingRatio: 1,
+        creationDate: expect.any(Number),
+      }),
+    );
+    expect(logger.warn).not.toHaveBeenCalled();
+  });
+
+  it('produces a warning when the metric value is non-numeric', async () => {
+    // @ts-ignore
+    await ldc.identify(carContext);
+    // @ts-ignore
+    ldc.track('the-event', { the: 'data' }, '12');
+
+    expect(logger.warn).toHaveBeenCalled();
+    expect(logger.warn).toHaveBeenCalledWith(
+      expect.stringMatching(/was called with a non-numeric/),
+    );
```

Comment:
The second assertion implies the first, so we can safely remove the first assertion.

```suggestion
    expect(logger.warn).toHaveBeenCalledWith(
      expect.stringMatching(/was called with a non-numeric/),
    );
```